lib/mindwendel_web/plugs/set_response_header_content_security_policy.ex

```diff
@@ -0,0 +1,74 @@
+defmodule Mindwendel.Plugs.SetResponseHeaderContentSecurityPolicy do
+  @content_secrity_policy_response_header_key "content-security-policy"
+
+  # @impl true
+  def init(opts) do
+    opts
+  end
+
+  # @impl true
+  def call(%Plug.Conn{} = conn, _opts) do
+    Plug.Conn.put_resp_header(
+      conn,
+      @content_secrity_policy_response_header_key,
+      content_secrity_policy_directives()
+    )
+  end
+
+  def content_secrity_policy_directives() do
+    # Usually, the csp directive `connect-src 'self'` is enough,
+    # see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/connect-src .
+    #
+    # However, the `connect-src 'self'` does not resolve to websocket schemes in all browsers, e.g. Safari,
+    # see https://github.com/w3c/webappsec-csp/issues/7 .
+    #
+    # Therefore, we need to explicitly add the allowed websocket uris here so that live views will work in safari in combination with CSP policies.
+    # e.g. `connect-src ws://localhost:*` or `connect-src wss://#{@host}` .
+    "default-src 'none' ; " <>
+      "connect-src 'self' #{
+        %URI{
+          scheme: get_websocket_scheme(),
+          host: get_host(),
+          port: get_port()
+        }
+      } ; " <>
+      "font-src    'self' ; " <>
+      "frame-src   'self' ; " <>
+      "img-src     'self' data: ; " <>
+      "script-src  'self' 'unsafe-eval' ; " <>
```

Comment:
do we really need unsafe eval? Fro mozilla docs:

Allow use of dynamic code [eval](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval)uation such as eval, [setImmediate](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate) Non-Standard , and window.execScript

That sounds like a potential js injection issue to me (if someone posts javascript).

Reply:
At the moment, we do need this. I do not know exactly why. Maybe because of live view implementation. But you are right. We should investigate this in a separate issue.